radicle-surf/src/git/repo.rs

```diff
@@ -252,6 +257,28 @@ impl Repository {
         Ok(Directory::root(tree.id().into()))
     }
 
+    /// Returns a [Directory] for `path` in `commit`.
+    pub fn directory<C: ToCommit>(&self, commit: C, path: &str) -> Result<Directory, Error> {
```

Comment:
I would change the `path` parameter to be more general in this case:
```suggestion
    pub fn directory<C: ToCommit, P: AsRef<Path>>(&self, commit: C, path: &P) -> Result<Directory, Error> {
```